src/main/java/it/gov/pagopa/apiconfig/core/service/CdiService.java

```diff
@@ -138,23 +140,29 @@ public void createCdi(MultipartFile file) {
     // map file into model class
     CdiXml xml = mapXml(file, CdiXml.class);
 
-    var psp = getPspIfExists(xml.getIdentificativoPSP());
+    // PAGOPA-936: skip the CDI creation for the CHARITY_PREFIX:
+    // it was chosen to use only the check on the 'identificativoPSP' because the one on the
+    // 'identificativoFlusso' is unnecessary
+    if (!xml.getIdentificativoPSP().startsWith(CHARITY_PREFIX)) {
```

Comment:
Seems all good, but I have a doubt: if the passed XML is related to a charity PSP, the controller will returns "201 CREATED" as response. Is this behavior correct or is better to have another kind of response (i.e. "400 BAD REQUEST" or other) that explain that is not possible to analyze XML from charity PSP?

Reply:
I too had the same doubt. Not being a real mistake I opted for the filter only but, in fact, it could be misleading.
@pasqualespica what do you think is the best behavior?

Reply:
right , in this case we should return something like https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422